%YAML:1.0

# Project
project_name: "lvi_sam"

#common parameters
imu_topic: "/imu/data"
image_topic: "/image_proc_resize/image"
point_cloud_topic: "lvi_sam/lidar/deskew/cloud_deskewed"

# Lidar Params
use_lidar: 1                     # whether use depth info from lidar or not
lidar_skip: 3                    # skip this amount of scans
align_camera_lidar_estimation: 0 # align camera and lidar estimation for visualization

# lidar to camera extrinsic (YPR = ZYX)
lidar_to_cam_tx: -0.13
lidar_to_cam_ty: -0.02
lidar_to_cam_tz: -0.18
lidar_to_cam_rx: -1.570793
lidar_to_cam_ry: 0
lidar_to_cam_rz: -1.5707963

# lidar to imu extrinsic 
lidar_to_imu_tx: -0.3787
lidar_to_imu_ty: 0.0
lidar_to_imu_tz: -0.6945
lidar_to_imu_rx: 0.0
lidar_to_imu_ry: 0.0
lidar_to_imu_rz: 0.0

# camera model (new camera)
model_type: PINHOLE
camera_name: camera

# Mono camera config
image_width: 480
image_height: 270
distortion_parameters:
   k1: 0.0
   k2: 0.0
   p1: 0.0
   p2: 0.0
projection_parameters:
   fx: 230.39028930664062
   fy: 230.31454467773438
   cx: 239.93666076660156
   cy: 136.52784729003906


#imu parameters       The more accurate parameters you provide, the worse performance
acc_n: 0.08         # accelerometer measurement noise standard deviation.
gyr_n: 0.04         # gyroscope measurement noise standard deviation.
acc_w: 0.00004        # accelerometer bias random work noise standard deviation.
gyr_w: 2.0e-6       # gyroscope bias random work noise standard deviation.
g_norm: 9.81007       #

# Extrinsic parameter between IMU and Camera.
estimate_extrinsic: 0   # 0  Have an accurate extrinsic parameters. We will trust the following imu^R_cam, imu^T_cam, don't change it.
                        # 1  Have an initial guess about extrinsic parameters. We will optimize around your initial guess.
                        # 2  Don't know anything about extrinsic parameters. You don't need to give R,T. We will try to calibrate it. Do some rotation movement at beginning. 
#Rotation from camera frame to imu frame, imu^R_cam
extrinsicRotation: !!opencv-matrix
   rows: 3
   cols: 3
   dt: d
   data: [ 0, 0, 1, 
           -1, 0, 0, 
           0, -1, 0]

#Translation from camera frame to imu frame, imu^T_cam
extrinsicTranslation: !!opencv-matrix
   rows: 3
   cols: 1
   dt: d
   data: [0.4681, 0.0, 0.2539] 

# # Rotation from imu frame to cam
# extrinsicRotation: !!opencv-matrix
#    rows: 3
#    cols: 3
#    dt: d
#    data: [ 0, -1, 0, 
#            0, 0, -1, 
#            1, 0, 0]
# # Translation from imu frame to cam
# extrinsicTranslation: !!opencv-matrix
#    rows: 3
#    cols: 1
#    dt: d
#    data: [0.0, 0.2539, -0.4681] 

#feature traker paprameters
max_cnt: 50            # max feature number in feature tracking
min_dist: 10            # min distance between two features 
freq: 0                # frequence (Hz) of publish tracking result. At least 10Hz for good estimation. If set 0, the frequence will be same as raw image 
F_threshold: 1.0        # ransac threshold (pixel)
show_track: 1           # publish tracking image as topic
equalize: 1             # if image is too dark or light, trun on equalize to find enough features
fisheye: 0              # if using fisheye, trun on it. A circle mask will be loaded to remove edge noisy points

#optimization parameters
max_solver_time: 0.04   # max solver itration time (ms), to guarantee real time
max_num_iterations: 8   # max solver itrations, to guarantee real time
keyframe_parallax: 10.0  # keyframe selection threshold (pixel)

#unsynchronization parameters
estimate_td: 0           # online estimate time offset between camera and imu
td: 0                    # initial value of time offset. unit: s. readed image clock + td = real image clock (IMU clock)

#rolling shutter parameters
rolling_shutter: 0       # 0: global shutter camera, 1: rolling shutter camera
rolling_shutter_tr: 0    # unit: s. rolling shutter read out time per frame (from data sheet). 

#loop closure parameters
loop_closure: 1                    # start loop closure
skip_time: 0.0
skip_dist: 0.0
debug_image: 0                      # save raw image in loop detector for visualization prupose; you can close this function by setting 0 
match_image_scale: 0.5
vocabulary_file: "/config/brief_k10L6.bin"
brief_pattern_file: "/config/brief_pattern.yml"